packages/core/src/routes/connector.ts

```diff
@@ -0,0 +1,86 @@
+import { Connectors } from '@logto/schemas';
+import { boolean, object, string } from 'zod';
+
+import { getConnectorInstanceById } from '@/connectors';
+import RequestError from '@/errors/RequestError';
+import { getAllConnectorInfo, getConnectorInfoById } from '@/lib/connectors';
+import koaGuard from '@/middleware/koa-guard';
+import { updateConnector } from '@/queries/connector';
+
+import { AuthedRouter } from './types';
+
+export default function connectorRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/connectors', async (ctx, next) => {
+    ctx.body = await getAllConnectorInfo();
+
+    return next();
+  });
+
+  router.get(
+    '/connectors/:id',
+    koaGuard({ params: object({ id: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+      ctx.body = getConnectorInfoById(id);
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/connectors/:id/enabled',
+    koaGuard({ params: object({ id: string().min(1) }), body: object({ enabled: boolean() }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+        body: { enabled },
+      } = ctx.guard;
+      if (typeof enabled === 'boolean') {
```

Comment:
1. again, a falsy-and-fail check will be more clear
```ts
if (!true) {
  throw new Error();
}
```
2. we already guarded `enabled` as boolean, why do we need to perform type checking here?